From 3c39af873af037149cdd8c57f9cc98b802fd27ac Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Tue, 21 Jul 2026 01:40:10 +0000
Subject: [PATCH 06274/15333] ASoC: qcom: storm: use dev instead of card on
 storm_parse_of()

storm_parse_of() can be processed without using *card.
No functional change, but is preparation for Card capsuling.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Reviewed-by: Srinivas Kandagatla <srinivas.kandagatla@oss.qualcomm.com>
Link: https://patch.msgid.link/87mrvli0hx.wl-kuninori.morimoto.gx@renesas.com
Signed-off-by: Mark Brown <broonie@kernel.org>
---
 sound/soc/qcom/storm.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/sound/soc/qcom/storm.c b/sound/soc/qcom/storm.c
index 1e0eda8c24c4..6b42fbfd27ca 100644
--- a/sound/soc/qcom/storm.c
+++ b/sound/soc/qcom/storm.c
@@ -64,21 +64,21 @@ static struct snd_soc_dai_link storm_dai_link = {
 	SND_SOC_DAILINK_REG(hifi),
 };
 
-static int storm_parse_of(struct snd_soc_card *card)
+static int storm_parse_of(struct device *dev)
 {
-	struct snd_soc_dai_link *dai_link = card->dai_link;
-	struct device_node *np = card->dev->of_node;
+	struct snd_soc_dai_link *dai_link = &storm_dai_link;
+	struct device_node *np = dev->of_node;
 
 	dai_link->cpus->of_node = of_parse_phandle(np, "cpu", 0);
 	if (!dai_link->cpus->of_node) {
-		dev_err(card->dev, "error getting cpu phandle\n");
+		dev_err(dev, "error getting cpu phandle\n");
 		return -EINVAL;
 	}
 	dai_link->platforms->of_node = dai_link->cpus->of_node;
 
 	dai_link->codecs->of_node = of_parse_phandle(np, "codec", 0);
 	if (!dai_link->codecs->of_node) {
-		dev_err(card->dev, "error getting codec phandle\n");
+		dev_err(dev, "error getting codec phandle\n");
 		return -EINVAL;
 	}
 
@@ -106,7 +106,7 @@ static int storm_platform_probe(struct platform_device *pdev)
 	card->dai_link	= &storm_dai_link;
 	card->num_links	= 1;
 
-	ret = storm_parse_of(card);
+	ret = storm_parse_of(&pdev->dev);
 	if (ret) {
 		dev_err(&pdev->dev, "error resolving dai links: %d\n", ret);
 		return ret;
-- 
2.55.0

